app/dashboard/idena_utils.py

```diff
@@ -100,12 +110,21 @@ def next_validation_time():
     return idena_validation_time
 
 def get_idena_status(address):
-    url = f'https://api.idena.io/api/Identity/{address}'
-    r = requests.get(url).json()
+    status =  None
 
-    if 'error' in r:
-        status = 'Not validated'
-    else:
-        status = r['result']['state']
+    if settings.IDENA_RPC_NODE:
+        try:
+            status = idena.client.dna.get_identity(address).state
+        except idena.exceptions.IdenaException as ex:
+            logging.warn('Idena node error: {ex}')
+
+    if status is None: 
+        url = f'https://api.idena.io/api/Identity/{address}'
+        r = requests.get(url).json()
+
+        if 'error' in r:
+            status = 'Not validated'
+        else:
+            status = r['result']['state']
```

Comment:
Why not use `r.get('result').get('state')` should `result` fail to be resolved, so it can fail silently, since we are already handling the worse case scenario on L130.

Reply:
Definitely, that is better, thank you for mentioning.